### PR TITLE
WIP: Immediate changes: Add net6 TFM to public nuget packages

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

For ADSK devs please see:
https://wiki.autodesk.com/pages/viewpage.action?pageId=1788742002

Goal of this specific PR is to support net6 consumers applications that need to compile against our public nugets immediately.

This PR does the following:
* updates all .nuspec files with a general net6 tfm (except for those that support net std 2.0)
* ~~adds a new nuget package, DynamoCoreNodesWPF~~
* reorganizes some binaries in the nuget packages to better reflect their dependency structure and use cases.
    * cross nuget dependencies have been added so that users should not have to add any new references themselves to get the binaries they were previously using.
* support all currently known dynamo player reference use cases.
* generates xml doc files for most of the binaries we package, and pack them into the nugets.
* update some descriptions.
* splits `UnitsUI.dll` into 2 dlls - `UnitsUI` and `UnitsNodeModels`, `Units.dll`'s csproj is renamed `UnitsCore` to better reflect that it contains core forge types which are useful for ZT nodes, but not NodeModels. I wanted to avoid bundling these Units  NodeModel nodes with the core types because this way, ZT clients don't need to reference/pull CoreNodeModels. I've used type forwarding for public types, but will need to do some quick migration tests with previously created graphs.

### TODO / open questions
- [ ] should  WPF packages use TFM `net6-windows`?
- [x] uncomment some file elements when all net6 binaries actually exist
- [ ] manually publish nugets, since atm master-15 does not build net6 .slns.
- [ ] review WIKI linked above with team, and file followup tasks after discussion for future medium and long term work.
- [ ] Ensure that the UI units nodes still load /migrate correctly and icons load correctly still works.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Libraries/DesignScriptBuiltin/DesignScriptBuiltin.csproj
Comment (pinzart90) [inline src/Libraries/DesignScriptBuiltin/DesignScriptBuiltin.csproj:12]: if "GenerateDocumentationFile" should be true for all projects, then just add it to the CS_SDK.props file

File: tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec
Comment (pinzart90) [inline tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:41]: one strategy I used in the past (to make sure I got the dependencies correct) was to generate packages (and nuspecs) for every package via csproj automation and merging them manually in the final nuspecs.

Comment (mjkkirschner) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:41]: will take a look at this - I am not convinced we actually want a fully transitive list of deps given the use cases we actually provide.

AFAIK these packages are not used for actually shipping any of our binaries, OR our deps - just for compiling against them. In fact, if devs ship these binaries it will cause conflicts when dyn packages/extensions are loaded into Dynamo/Revit etc.

Though - you could make the argument we don't really provide a public nuget for actually integrating and shipping Dynamo into a another project - but that seems to be a different missing package - or require a total rethink of these packages.

Comment (mjkkirschner) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:41]: @pinzart90 do you know if there is a way to provide a default package reference that excludes all binaries from being consumed? Or should we provide some docs on how to reference these packages with that snippet?

File: tools/NuGet/template-nuget/DynamoVisualProgramming.Tests.nuspec
Comment (mjkkirschner) [inline tools/NuGet/template-nuget/DynamoVisualProgramming.Tests.nuspec:33]: wait until this gets merged.

File: tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoCoreNodesWPF.nuspec
Comment (mjkkirschner) [inline tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoCoreNodesWPF.nuspec:43]: can uncomment now

File: tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoCoreNodesWPF.nuspec
Comment (mjkkirschner) [inline tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoCoreNodesWPF.nuspec:45]: wait until merged.

File: tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec
Comment (pinzart90) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:41]: https://learn.microsoft.com/en-us/nuget/create-packages/creating-a-package#from-a-convention-based-working-directory
You can also control compile/runtime references in the PackageReference syntax

Comment (pinzart90) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:41]: If it is just about compiling, I think dotnet6 has a build property that allows a project to produce reference only assemblies (just the public APIs) 
https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#producereferenceassemblyinoutdir
We could include them only in the ref/{tfm} subfolder so that they are only usable at compile time (I have not tested this yet)

Comment (mjkkirschner) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:41]: interesting idea @pinzart90 - I'll try that, it seems less flexible, but less likely to cause issues for clients, and more idiomatic given how we expect most users to actually use these packages. This is a repeated issue I see over and over on the forum etc - where users need to make copy local false on all our binaries.

Maybe the `DVP.Core` package and `DVP.Tests` package could continue using the `lib` folder as `DVP.Core` can be used to integrate Dynamo just the core of Dynamo.... I think.

FYI @twastvedt @QilongTang for feedback on this idea.

Anyhow - what do you think about merging this without the discussed `lib` -> `ref`, and I can tackle that in a separate PR / followup, I think it's a very good idea to test out and get feedback internally and externally from clients - but I'd like to get this merged so we can unblock player, revit etc.

Comment (pinzart90) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:41]: Sounds good to me

Comment (twastvedt) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:41]: A separate PR to put dlls in `ref` makes sense to me. Not sure about whether they should be reference assemblies or not though. That does seem typical, and would make for somewhat smaller nugets (not critical, I think). However, then consumers would lose the ability to see decompiled source logic when they click "Go to definition" on a Dynamo API, which can be handy.

Comment (mjkkirschner) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:41]: I'm missing something.
1. Why would this make smaller nugets?
2. Why would consumers lose the decompile source option in vs? Is this documented somewhere?

My understanding -which could be incorrect, was that using the `ref` folder just kept the binaries from being copied to the output path of the consuming project.

File: tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec
Comment (aparajit-pratap) [inline tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:35]: Any answer to this question?

File: tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoCoreNodesWPF.nuspec
Comment (aparajit-pratap) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoCoreNodesWPF.nuspec:45]: Any reason for not including other WPF-based nodes in `Watch3DNodeModelsWPF`, `PythonNodeModelsWPF`, `GeometryUIWPF`, etc.?

File: tools/NuGet/template-nuget/DynamoVisualProgramming.WpfUILibrary.nuspec
Comment (aparajit-pratap) [inline tools/NuGet/template-nuget/DynamoVisualProgramming.WpfUILibrary.nuspec:38]: Is there an analagous nuget package like this one for headless applications that references DynamoCore and DynamoCoreNodes packages?

File: tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec
Comment (twastvedt) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:41]: Sorry, yes, confusing. We can manually put any dll we like in the ref folder. But Microsoft also has the concept of a reference assembly. In idiomatic nuget packages (I think) the `ref` folder contains [reference assemblies](https://learn.microsoft.com/en-us/dotnet/standard/assembly/reference-assemblies). These assemblies have the implementation and private properties compiled out.

Comment (mjkkirschner) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:41]: @twastvedt thanks very interesting - I will note that this should be tested when I/we work on this. From my reading- I think that since we are not actually generating reference assemblies, we'll get both "features" that we want - though it won't be completely idiomatic.
1.  users's won't have to manually remove all the dynamo binaries/deps from their output folders
2. the assemblies will still contain implementation.

I think this is a good compromise for most of these packages. Less convinced for dynamo core package.

Anyway, thanks for the info.

File: tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoCoreNodesWPF.nuspec
Comment (mjkkirschner) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoCoreNodesWPF.nuspec:45]: there just wasn't a use case for them, and actually after @twastvedt's latest comment on the wiki I think I can also remove the `UnitsUI` nuget as well - so I will need to think about where to put `CoreNodeModelsWpf` a bit more.

I don't think we should ship these UI binaries unless absolutely needed especially because we break them and don't semantically version them.

File: tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec
Comment (pinzart90) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:35]: This is the full dependency list generated by dotnet
```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd">
  <metadata>
    <id>DynamoCore</id>
    <version>1.0.0</version>
    <authors>DynamoCore</authors>
    <description>Package Description</description>
    <dependencies>
      <group targetFramework=".NETFramework4.8">
        <dependency id="DynamoUtilities" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="GraphLayout" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="ProtoAssociative" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="ProtoCore" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="ProtoImperative" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="ProtoScript" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="DesignScriptBuiltin" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="DynamoUnits" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="VMDataBridge" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="DynamoServices" version="1.0.0" exclude="Build,Analyzers" />
        <dependency id="Autodesk.IDSDK" version="1.1.6" exclude="Build,Analyzers" />
        <dependency id="DynamoVisualProgramming.Analytics" version="3.0.1.740" exclude="Build,Analyzers" />
        <dependency id="DynamoVisualProgramming.LibG_228_0_0" version="2.18.0.1400" exclude="Build,Analyzers" />
        <dependency id="DynamoVisualProgramming.LibG_229_0_0" version="2.18.0.1400" exclude="Build,Analyzers" />
        <dependency id="Greg" version="2.5.0.5076" exclude="Build,Analyzers" />
        <dependency id="Lucene.Net" version="4.8.0-beta00016" exclude="Build,Analyzers" />
        <dependency id="Lucene.Net.Analysis.Common" version="4.8.0-beta00016" exclude="Build,Analyzers" />
        <dependency id="Lucene.Net.QueryParser" version="4.8.0-beta00016" exclude="Build,Analyzers" />
        <dependency id="Newtonsoft.Json" version="13.0.1" exclude="Build,Analyzers" />
        <dependency id="RestSharp" version="106.12.0" exclude="Build,Analyzers" />
      </group>
    </dependencies>
    <frameworkAssemblies>
      <frameworkAssembly assemblyName="Microsoft.CSharp" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="Microsoft.Win32.Primitives" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.AppContext" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Configuration" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Console" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Diagnostics.Tracing" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Globalization.Calendars" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.IO.Compression" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.IO.Compression.ZipFile" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.IO" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.IO.FileSystem" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.IO.FileSystem.Primitives" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Linq" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Linq.Expressions" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Net.Http" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Net.Sockets" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Reflection" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Runtime" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Runtime.Extensions" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Runtime.InteropServices" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Runtime.InteropServices.RuntimeInformation" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Security.Cryptography.Algorithms" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Security.Cryptography.Encoding" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Security.Cryptography.Primitives" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Security.Cryptography.X509Certificates" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Text.RegularExpressions" targetFramework=".NETFramework4.8" />
      <frameworkAssembly assemblyName="System.Xml.ReaderWriter" targetFramework=".NETFramework4.8" />
    </frameworkAssemblies>
  </metadata>
</package>
```
I am not sure anyone will find the partial list useful.

Comment (mjkkirschner) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:35]: 1. hmm thanks @pinzart90 - can you tell me how you generated that list?

2. If someone will find the partial list useful or not: that ultimately depends on what the use case is, and for simply writing nodeModel nodes - users really only need `system.collections.immutable` and `newtonsoft` references present to compile correctly.

I am not sure why users would need Nunit present for DynamoCore consumption.

Again - if someone was actually going to use this package to embed DynamoModel into something then they would definitely be helped by having your full list.

BUT having that full list will make the first use case of writing nodes for Dynamo very difficult.

3. so here is what I am thinking:

* I do not wish to include this full list until these packages are transitioned to `ref` folder.
* In the next PR I will look at moving all current packages to ref, and including full dep lists so that all reference assemblies that could be used in public APIs are available at compile time.
* I will file a task to consider the embedding of `DynamoModel` as a separate use case from `DynamoVisualProgrmaming.Core` and to create a separate package with full dep list and using the `lib` folder. We can prioritize that later.

File: tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoCoreNodesWPF.nuspec
Comment (aparajit-pratap) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoCoreNodesWPF.nuspec:45]: So it looks like as per Trygve's comment we need to split UnitsUI and add its nodemodels half into the DynamoCoreNodes package.

File: tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec
Comment (mjkkirschner) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:52]: I just didn't think it was a useful API for this use case - but I'll err on the side of consistency - added.

File: tools/NuGet/template-nuget/DynamoVisualProgramming.WpfUILibrary.nuspec
Comment (mjkkirschner) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.WpfUILibrary.nuspec:42]: No, I don't think so, I will add it to the future use cases to support. I will be filing some followup tasks for sure - because I think at the moment even the current DVP.Core package won't work 100% for a true headless integration.

File: src/Libraries/DynamoUnits/Properties/AssemblyInfo.cs
Comment (pinzart90) [inline src/Libraries/DynamoUnits/Properties/AssemblyInfo.cs:19]: so we are keeping this assembly (DynamoUnits) around just for type forwarding right ?

Comment (pinzart90) [reply src/Libraries/DynamoUnits/Properties/AssemblyInfo.cs:19]: sorry this change is a bit confusing.
So the old DynamoUnits project will now be UnitsCore.csproj but it will produce the same dll as before (for compatibility sake and typeforwarding)
Is that correct ?

File: tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec
Comment (pinzart90) [inline tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:38]: Why do we need NUnit in the Core package ?
We already have another Test package...

File: src/Libraries/DynamoUnits/Properties/AssemblyInfo.cs
Comment (mjkkirschner) [reply src/Libraries/DynamoUnits/Properties/AssemblyInfo.cs:19]: hey @pinzart90 it's pretty confusing yes, let me try to explain.

`DynamoUnits.dll` (previously Units.csproj) contains the new ZT nodes that use the Forge types. It also exposes those Forge types with Dynamo wrappers to users can write ZT nodes that accept and return those types. For example `DynamoUnits.Unit`.

I renamed this csproj to `UnitsCore` - but I have no desire to mess with the assembly name at this time for the reasons you mentioned - lets do it at 2.20 or 3.0 etc.

`UnitsUI.dll` contained both the WPF controls and the node models nodes for some more complex UI units nodes. I've split this into 
`UnitsUI.dll` and `UnitsNodeModels.dll`

I originally put the nodeModel nodes into DynamoUnits.dll - but these nodes have dependencies on `CoreNodeModels` and I didn't want to force all ZT node authors to reference/pull those packages just to get access to the core `Unit` type for their nodes.

I hope that explanation helps.

File: tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec
Comment (mjkkirschner) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec:38]: Yep, I agree, I have it on my list to try building the samples repo after I remove this dep (it's just existing) - but I thought I might do it in a followup PR.

**Comment by mjkkirschner** [comment]
All stable tests are passing but I have some conflicts to resolve, as well as some updates to make after @aparajit-pratap 's latest PR to nuspecs.

**Comment by mjkkirschner** [comment]
> @mjkkirschner can you explain what you mean by `I wanted to avoid bundling these Units NodeModel nodes with the core types because this way, ZT clients don't need to reference/pull CoreNodeModels.`

Hey @aparajit-pratap 

I've tried to explain the issue here:
https://github.com/DynamoDS/Dynamo/pull/14137#discussion_r1262981262

the crux of your question is that UnitNodeModels (these UI unit nodes) - have dependencies on `CoreNodeModels` - but the Forge unit nodes (ZT) don't have such a dep, and I wanted to keep that assembly as dependency free as possible because the core types (unit, quantity, etc) are useful when writing ZT nodes.

File: src/DynamoCore/DynamoCore.csproj
Comment (aparajit-pratap) [inline src/DynamoCore/DynamoCore.csproj:76]: Was this an extraneous dependency that wasn't being used by DynamoCore?

File: src/Libraries/CoreNodes/CoreNodes.csproj
Comment (aparajit-pratap) [inline src/Libraries/CoreNodes/CoreNodes.csproj:58]: Same question here.

File: src/DynamoCore/DynamoCore.csproj
Comment (mjkkirschner) [reply src/DynamoCore/DynamoCore.csproj:76]: yep - there is a new feature (new to me?) in vs 2022 - remove unused references.

File: src/Libraries/CoreNodes/CoreNodes.csproj
Comment (mjkkirschner) [reply src/Libraries/CoreNodes/CoreNodes.csproj:58]: yep - there is a new feature (new to me?) in vs 2022 - remove unused references.

File: test/TestUINodes/TestUINodes.csproj
Comment (aparajit-pratap) [inline test/TestUINodes/TestUINodes.csproj:43]: This is to prevent these binaries from being copied into the `nodes` subfolder?

File: tools/NuGet/template-artifactory/DynamoVisualProgramming.DynamoSamples.nuspec
Comment (aparajit-pratap) [inline tools/NuGet/template-artifactory/DynamoVisualProgramming.DynamoSamples.nuspec:16]: I deliberately added this comment as the other nuspecs reference files from the current directory without having to reference them from this long relative path. I'm still not clear why there's a difference. See [here](https://github.com/DynamoDS/Dynamo/blob/5634759b89a76b668c1e815f9d3be0de433d2f5b/tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec#L35).

File: test/TestUINodes/TestUINodes.csproj
Comment (mjkkirschner) [reply test/TestUINodes/TestUINodes.csproj:43]: yes, I tried using @pinzart90's suggestion of using 
```
<ItemDefinitionGroup>
    <ReferenceCopyLocalPaths>
	  <!--Copy all assembly references to the OutputPath parent dir (/nodes/../) -->
	  <DestinationSubDirectory>..\</DestinationSubDirectory>
    </ReferenceCopyLocalPaths>
  </ItemDefinitionGroup>
```

but I don't think this is the correct solution - and I will file a followup to do something more idiomatic for the other projects that use it.

For example, it does not work for this TestUINodes.csproj project - because this project references other binaries in the `nodes` folder - it then copies those binaries, not just their deps, to the `bin` folder - so we end up with things like `UnitsUI` in both `bin` and `nodes`.

File: tools/NuGet/template-artifactory/DynamoVisualProgramming.Tests.nuspec
Comment (aparajit-pratap) [inline tools/NuGet/template-artifactory/DynamoVisualProgramming.Tests.nuspec:34]: I added this nuspec for tests for net6: https://github.com/DynamoDS/Dynamo/blob/master/tools/NuGet/template-net60-core-runtime/DynamoVisualProgramming.TestsNet6.nuspec

Comment (mjkkirschner) [reply tools/NuGet/template-artifactory/DynamoVisualProgramming.Tests.nuspec:34]: @aparajit-pratap - in this PR I am trying to use a single nuspec for the both net6 and net48 as that is idiomatic. I don't think there will be an issue doing it this way, as these are quite small.

We'll use two nuspecs for the runtimecore - because its hundreds of megs.

Is that ok with you, so I will remove this separate nuspec?

Comment (aparajit-pratap) [reply tools/NuGet/template-artifactory/DynamoVisualProgramming.Tests.nuspec:34]: I guess that's fine by me, it's just a little confusing, that's all.

Comment (aparajit-pratap) [reply tools/NuGet/template-artifactory/DynamoVisualProgramming.Tests.nuspec:34]: Never mind my last comment, I suppose this was the original intent of this PR, which was to add net6 TFM to public nugets, I forgot.

**Review by aparajit-pratap — APPROVED** [review]
@mjkkirschner so besides making another change to remove `https://github.com/DynamoDS/Dynamo/blob/master/tools/NuGet/template-net60-core-runtime/DynamoVisualProgramming.TestsNet6.nuspec`, LGTM.

**Comment by mjkkirschner** [comment]
I'm going to merge this before any more changes come in. I think the deps are mostly in order - @sm6srw I did notice your last PR brought some testing deps into the `/nodes` dir, so I cleaned that up here.

@QilongTang - you should be able to use these nuspecs to publish net6 public packages in your current work.

**Pull request merged**
